src/pkg/services/m365/api/item_pager.go

```diff
@@ -172,12 +173,15 @@ func deltaEnumerateItems[T any](
 // shared enumeration runner funcs
 // ---------------------------------------------------------------------------
 
+type addedAndRemovedHandler[T any] func(items []T) ([]string, []string, error)
+
 func getAddedAndRemovedItemIDs[T any](
```

Comment:
I'm a little concerned by the growing number of parameters we're gathering here. We may need to refactor this at some point in part to help that issue but also because I've been trying to get some stuff up and running for Exchange kopia assisted incrementals and I've found events doesn't fit the expected pattern. It doesn't return mod times in delta tokens which throws off some of the other logic in this function

Reply:
I actually just put this through a fairly heavy refactor and this is, I think, as tight as we're going to make the func.  Then again, this is unexported, and should only be getting called through funcs [like this](https://github.com/alcionai/corso/blob/main/src/pkg/services/m365/api/mail_pager.go#L282), which should provide a nicer api to interface with.  Additional parameters like this shouldn't upset anything that higher levels are trying to do.

What it sounds like you're bringing up isn't a concern with the function handle, so much as how we're natively reducing the data to ids, and that's causing you to slice out modified times.  Is that right?  If so, it should be a fairly light lift to extend that dataset.  And since I've already got myself elbow-deep in the code, I'd be happy to make the change.

Reply:
>What it sounds like you're bringing up isn't a concern with the function handle, so much as how we're natively reducing the data to ids, and that's causing you to slice out modified times

Unfortunately it's not quite that simple :( Graph's delta results don't have a mod time returned in them which leaves us up a creek a bit when it comes to following existing code paths we have. I was hoping that I could just switch this function to return `map[string]time.Time` for added items (and other functions that call this) so that we get that into the collection. We could then leverage that to delay fetching item data until we know we need it since the file path and mod time is sufficient to get kopia's cached item checks working

However, since we don't have mod times for some items we either need to split events off to use it's own stuff or figure out a more flexible way to deal with things here. I'll keep thinking on it tomrrow

Reply:
> However, since we don't have mod times for some items we either need to split events off to use it's own stuff or figure out a more flexible way to deal with things here. I'll keep thinking on it tomrrow

Incidentally, this shares some overlap with channel messages.  The messages themselves return their mod time.  However, since we're packing the replies into the message, that mod time won't reflect whether the entire message thread has had a more recent addition.  And at a surface glance, there's no easy way to get that info without enumerating all replies.

We might need to revisit the approach at a broad scale, beyond just the one or two services using it.  Maybe we can sync tomorrow and noodle on it together.